plugin/pkg/seed/validator/admission.go

```diff
@@ -0,0 +1,160 @@
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return New()
+	})
+}
+
+// ValidateSeed contains listers and and admission handler.
+type ValidateSeed struct {
+	*admission.Handler
+	seedLister         corelisters.SeedLister
+	shootLister        corelisters.ShootLister
+	backupBucketLister corelisters.BackupBucketLister
+	readyFunc          admission.ReadyFunc
+}
+
+var (
+	_ = admissioninitializer.WantsInternalCoreInformerFactory(&ValidateSeed{})
+
+	readyFuncs = []admission.ReadyFunc{}
+)
+
+// New creates a new ValidateSeed admission plugin.
+func New() (*ValidateSeed, error) {
+	return &ValidateSeed{
+		Handler: admission.NewHandler(admission.Delete),
+	}, nil
+}
+
+// AssignReadyFunc assigns the ready function to the admission handler.
+func (v *ValidateSeed) AssignReadyFunc(f admission.ReadyFunc) {
+	v.readyFunc = f
+	v.SetReadyFunc(f)
+}
+
+// SetInternalCoreInformerFactory gets Lister from SharedInformerFactory.
+func (v *ValidateSeed) SetInternalCoreInformerFactory(f coreinformers.SharedInformerFactory) {
+	seedInformer := f.Core().InternalVersion().Seeds()
+	v.seedLister = seedInformer.Lister()
+
+	shootInformer := f.Core().InternalVersion().Shoots()
+	v.shootLister = shootInformer.Lister()
+
+	backupBucketInformer := f.Core().InternalVersion().BackupBuckets()
+	v.backupBucketLister = backupBucketInformer.Lister()
+
+	readyFuncs = append(readyFuncs, seedInformer.Informer().HasSynced, shootInformer.Informer().HasSynced, backupBucketInformer.Informer().HasSynced)
+}
+
+// ValidateInitialization checks whether the plugin was correctly initialized.
+func (v *ValidateSeed) ValidateInitialization() error {
+	if v.seedLister == nil {
+		return errors.New("missing seed lister")
+	}
+	if v.shootLister == nil {
+		return errors.New("missing shoot lister")
+	}
+	if v.backupBucketLister == nil {
+		return errors.New("missing backupbucket lister")
+	}
+	return nil
+}
+
+var _ admission.ValidationInterface = &ValidateSeed{}
+
+// Validate validates the Seed details against existing Shoots and BackupBuckets
+func (v *ValidateSeed) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	// Wait until the caches have been synced
+	if v.readyFunc == nil {
+		v.AssignReadyFunc(func() bool {
+			for _, readyFunc := range readyFuncs {
+				if !readyFunc() {
+					return false
+				}
+			}
+			return true
+		})
+	}
+	if !v.WaitForReady() {
+		return admission.NewForbidden(a, errors.New("not yet ready to handle request"))
+	}
+
+	// Ignore all kinds other than Shoot
+	if a.GetKind().GroupKind() != core.Kind("Seed") {
+		return nil
+	}
+
+	// Ignore updates to status or other subresources
+	if a.GetSubresource() != "" {
+		return nil
+	}
+
+	seedName := a.GetName()
+	if len(seedName) == 0 {
+		return apierrors.NewInternalError(errors.New("empty Seed name"))
+	}
+
+	// Exit early if shoot spec hasn't changed
+	if a.GetOperation() != admission.Delete {
```

Comment:
Why do we have to verify this again? The admission plugin is only registered for `Delete`